tests/EvmAcceptanceTests/contracts/ERC20.sol

```diff
@@ -91,7 +91,7 @@ contract ZEVMToken is ERC20Interface, SafeMath {
  
     function transferFrom(address from, address to, uint tokens) external returns (bool success) {
         balances[from] = safeSub(balances[from], tokens);
-        allowed[from][msg.sender] = safeSub(allowed[from][msg.sender], tokens);
+        allowed[from][to] = safeSub(allowed[from][to], tokens);
```

Comment:
seems like we shouldn't touch it. see (this)[https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.7.3/contracts/token/ERC20/ERC20.sol#L158]

Reply:
Hm, what would be the solution in this case? Either the test is wrong or the contract code is wrong I guess?

Reply:
yep. after reading openzepplin, I realized that it should be like this:

```solidity
    it("Should be possible to transfer from one account to another [@transactional] xyz", async function () {
      const [owner, _, sender, spender] = await ethers.getSigners();

      // Fund the 2nd account first
      await contract.transfer(sender.address, 5_000);

      // Approve 30_000 for withdrawal by Acc #3
      await contract.connect(sender).approve(owner.address, 3_000);

      // Try transferring allowed amount
      expect(await contract.transferFrom(sender.address, spender.address, 2_999))
        .to.changeTokenBalances(contract, [sender.address, spender.address], [-2_999, +2_999])
        .emit(contract, "Transfer")
        .withArgs(sender.address, spender.address, 1_999);

      expect(await contract.allowance(sender.address, owner.address)).to.be.eq(1);
    });
```

Reply:
I checked your version and it still fails during eth_estimateGas (this is before transaction is made).

Reply:
LMTAL

Reply:
anyway, this change to the contract shouldn't have been committed. [It's not correct](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.7.3/contracts/token/ERC20/ERC20.sol#L158).